### ISSUE TITLE
Failure when no data_val

### ISSUE THREAD

**Issue by DrAnaximandre** [tboquet/python-alp#48]
When you fit an `experiment` without data_val you get a failure, at least with sklearn backend.

````
import alp
from alp.appcom.core import Experiment
from sklearn import cross_validation
from sklearn import datasets
from sklearn.linear_model import LogisticRegression
import numpy as np
np.random.seed(1234)

data = {"X": np.random.rand(10,3),"y":np.random.random_integers(0,1,10)}
lr = LogisticRegression()
expe = Experiment(lr)
expe.fit([data],[])
````

Traceback:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-2-834b1dd9453c> in <module>()
     11 lr = LogisticRegression()
     12 expe = Experiment(lr)
---> 13 expe.fit([data],[])
     14 

/root/python-alp-2/src/alp/appcom/core.pyc in fit(self, data, data_val, model, *args, **kwargs)
    109         """
    110         res = self._prepare_fit(model, data, data_val, generator=False,
--> 111                                 delay=False, *args, **kwargs)
    112         return res
    113 

/root/python-alp-2/src/alp/appcom/core.pyc in _prepare_fit(self, model, data, data_val, generator, delay, *args, **kwargs)
    334                 size_gen=size_gen,
    335                 generator=generator,
--> 336                 *args, **kwargs)
    337 
    338         return self._handle_results(res, delay)

/opt/conda/lib/python2.7/site-packages/celery/local.pyc in __call__(self, *a, **kw)
    186 
    187     def __call__(self, *a, **kw):
--> 188         return self._get_current_object()(*a, **kw)
    189 
    190     def __len__(self):

/opt/conda/lib/python2.7/site-packages/celery/app/task.pyc in __call__(self, *args, **kwargs)
    426             if self.__self__ is not None:
    427                 return self.run(self.__self__, *args, **kwargs)
--> 428             return self.run(*args, **kwargs)
    429         finally:
    430             self.pop_request()

/root/python-alp-2/src/alp/backend/sklearn_backend.pyc in fit(self, backend_name, backend_version, model, data, data_hash, data_val, size_gen, generator, *args, **kwargs)
    447                                           params_dump, data_hash,
    448                                           hexdi_m,
--> 449                                           *args, **kwargs)
    450 
    451         db.update({"_id": mod_id}, {'$set': res_dict})

/root/python-alp-2/src/alp/backend/common.pyc in train_pipe(train_f, save_f, model, data, data_val, generator, size_gen, params_dump, data_hash, hexdi_m, *args, **kwargs)
    194                              data_val, size_gen,
    195                              generator=generator,
--> 196                              *args, **kwargs)
    197     res_dict = {
    198         'iter_stopped': results['metrics']['iter'],

/root/python-alp-2/src/alp/backend/sklearn_backend.pyc in train(model, data, data_val, size_gen, generator, *args, **kwargs)
    287         data = [pickle.loads(d) for d in data]
    288 
--> 289     val_gen = check_gen(data_val)
    290 
    291     if val_gen > 0:

/root/python-alp-2/src/alp/appcom/utils.pyc in check_gen(iterable)
    236         True if the last object is a generator, False otherwise.
    237         """
--> 238     is_gen = (hasattr(iterable[-1], 'next') or
    239               hasattr(iterable[-1], '__next__'))
    240     is_gen += 'fuel' in repr(iterable[-1])

IndexError: list index out of range
```

I guess that adding a check that` data_val` exists, l289 in `sklearn_backend.py` should fix that.
Same for  l305 in  `keras_backend.py`

**Comment by DrAnaximandre** [comment]
Also add this case in tests.

**Comment by tboquet** [comment]
closed in #60.
